### PR TITLE
New Mexico Social Security income exemption

### LINKED ISSUE #2673: New Mexico Social Security income exemption
[Reference](https://nmonesource.com/nmos/nmsa/en/item/4340/index.do#!fragment/zoupio-_Toc140503690/BQCwhgziBcwMYgK4DsDWszIQewE4BUBTADwBdoAvbRABwEtsBaAfX2zgEYAWABgFYeAZgBsATh4BKADTJspQhACKiQrgCe0AOSapEQmFwJlqjdt37DIAMp5SAIQ0AlAKIAZZwDUAggDkAws5SpGAARtCk7BISQA)

### PR DESCRIPTION
**Pull request by PavelMakarchuk**
Fixes #2673

### Summary
:sparkles::page_facing_up::white_check_mark:

This pull request adds a new feature to the policy engine: the New Mexico social security income exemption. It updates the parameters, variables, and tests for the New Mexico income tax system, and adds a changelog entry to document the change.

> _New feature added_
> _`social_security_income`_
> _Exemption for NM_

### Walkthrough
*  Add a new feature to the policy engine: the New Mexico social security income exemption ([link](https://github.com/PolicyEngine/policyengine-us/pull/2674/files?diff=unified&w=0#diff-5885042762833a4aef51405a4036067e4e6aefa7a5091a2987d1d9d1fc2647e8R1-R4))
*  Define the income limit parameter for the exemption by filing status and year in `income_limit.yaml` ([link](https://github.com/PolicyEngine/policyengine-us/pull/2674/files?diff=unified&w=0#diff-0d90a71b014a0fbfa18166e2a274c52eda4eba4a2952f0219d938636d1a0bd3eR1-R28))
*  Implement the formula to calculate the exemption amount based on the social security income and the filing status of the tax unit in `nm_social_security_income_exemption.py` ([link](https://github.com/PolicyEngine/policyengine-us/pull/2674/files?diff=unified&w=0#diff-59ea77ba234a0a441ec4ea1925db89b961db0e1cf4dd09cab94dd3afe0bd6c72R1-R23))
*  Add four test cases to check the logic and output of the new feature in `nm_social_security_income_exemption.yaml` ([link](https://github.com/PolicyEngine/policyengine-us/pull/2674/files?diff=unified&w=0#diff-f4f24dff9f5d2649bb53c7571a6747de81079783484f5d82bd7baecc85cc80e8R1-R35))
*  Update the existing parameters for the low and middle income exemption in New Mexico for separate filers, based on the 2022 tax form packet, in `income_limit.yaml` and `income_threshold.yaml` ([link](https://github.com/PolicyEngine/policyengine-us/pull/2674/files?diff=unified&w=0#diff-4af9fc89b7fe93ec624e478e88be4a04095c46588c0935a692d23645230771d7L25-R25), [link](https://github.com/PolicyEngine/policyengine-us/pull/2674/files?diff=unified&w=0#diff-35daeabe04a672659f99a7baeddc13f223af789cfd61f48978056ba115676bf6L26-R26))

### CODE REVIEW

File: policyengine_us/variables/gov/states/nm/tax/income/exemptions/nm_social_security_income_exemption.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+    def formula(tax_unit, period, parameters):
+        social_security_income = tax_unit(
+            "tax_unit_taxable_social_security", period
+        )
+        filing_status = tax_unit("filing_status", period)
+        p = parameters(
+            period
+        ).gov.states.nm.tax.income.exemptions.social_security_income
+        income_eligible = (
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/nm/tax/income/exemptions/nm_social_security_income_exemption.py:20]:
```suggestion
        income_limit = p.income_limit[filing_status]
        agi = tax_unit("adjusted_gross_income", period)
        eligible = agi <= income_limit
```

File: policyengine_us/parameters/gov/states/nm/tax/income/exemptions/social_security_income/income_limit.yaml [hunk 2]
```diff
@@ -0,0 +1,28 @@
+description: New Mexico provides the secial security income exemption for filers with adjusted gross income at or below this amount.
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/nm/tax/income/exemptions/social_security_income/income_limit.yaml:1]:
```suggestion
description: New Mexico exempts taxable Social Security benefits for filers with adjusted gross income at or below this amount.
```

File: policyengine_us/parameters/gov/states/nm/tax/income/exemptions/social_security_income/income_limit.yaml [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+metadata:
+  label: New Mexico social security income exemption income limit
+  period: year
+  unit: currency-USD
+  breakdown:
+  - filing_status
+  reference:
+  - title: 2022 New Mexico personal income tax form packet ADJ-7 Exemption for social security income.
+    href: https://klvg4oyd4j.execute-api.us-west-2.amazonaws.com/prod/PublicFiles/34821a9573ca43e7b06dfad20f5183fd/1afc56af-ea90-4d48-82e5-1f9aeb43255a/PITbook2022.pdf#page=52
+  - title: New Mexico Statutes, 7-2-5.14. Exemption; social security income.
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/nm/tax/income/exemptions/social_security_income/income_limit.yaml:11]: That it came into effect in 2022 seems clear from the law, but could you cite a 2021 tax form too to show its absence?

File: policyengine_us/variables/gov/states/nm/tax/income/exemptions/nm_social_security_income_exemption.py [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+class nm_social_security_income_exemption(Variable):
+    value_type = float
+    entity = TaxUnit
+    label = "New Mexico social security income exemption"
+    defined_for = StateCode.NM
+    unit = USD
+    definition_period = YEAR
+
+    def formula(tax_unit, period, parameters):
+        social_security_income = tax_unit(
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/nm/tax/income/exemptions/nm_social_security_income_exemption.py:13]:
```suggestion
        # This is the amount of Social Security included in AGI.
        taxable_social_security = tax_unit(
```

**Review by MaxGhenis — CHANGES_REQUESTED** [review]

File: policyengine_us/variables/gov/states/nm/tax/income/exemptions/nm_social_security_income_exemption.py [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+    def formula(tax_unit, period, parameters):
+        # This is the amount of Social Security included in AGI.
+        social_security_income = tax_unit(
+            "tax_unit_taxable_social_security", period
+        )
+        filing_status = tax_unit("filing_status", period)
+        p = parameters(
+            period
+        ).gov.states.nm.tax.income.exemptions.social_security_income
+        income = tax_unit("nm_agi", period)
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/nm/tax/income/exemptions/nm_social_security_income_exemption.py:21]: Similar to the NM CTC, it only refers to AGI. Does that mean federal or NM? Or are they the same?

Comment (PavelMakarchuk) [reply policyengine_us/variables/gov/states/nm/tax/income/exemptions/nm_social_security_income_exemption.py]: Adjusted to the correct AGI
image

**Review by MaxGhenis — APPROVED** [review]

**Pull request merged**